doc/src/site/sphinx/experimental/rest_api_based_client.rst

```diff
@@ -0,0 +1,58 @@
+Rest API Based Client
+---------------------
+
+Supporting dynamic allocation in Sparkling Water is a long-term issue and is implemented in
+several stages. The first stage is replacing H2O node running inside of the Spark executor
+by lightweight REST API client.
+
+The first iteration of replacing the H2O Node by REST API client is currently tested only
+under PySparkling in External Backend Mode. At this stage we do not support using PySparkling
+Algo API in this new approach. After the PySparkling Algo API is migrated to the new approach
+as well, we switch to the REST API Based client approach by default for PySparkling in external
+backend.
+
+Starting Sparkling Water with the REST API Client
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To enable the REST API Based client, please start PySparkling with the spark option ``spark.ext.h2o.rest.api.based.client=true``.
+This ensures that there won't be any H2O client running on the Spark driver side in both manual and automatic
+start of external H2O backend.
+
+Changes on the API
+~~~~~~~~~~~~~~~~~~
+The API remains unaffected for the users of automatic mode of external backend.
+
+For the users of the manual backend we have simplified the configuration and there is no need to specify cluster size anymore in advance.
+Sparkling Water automatically discovers the cluster size. In particular ``spark.ext.h2o.external.cluster.size`` is not mandatory
+anymore for the users.
+
+The rest of the API also remains unaffected, except that the PySparkling Algo API is not currently supported in the
+first roll-out, but will follow up as soon as possible.
+
+Extended H2O Jars
+~~~~~~~~~~~~~~~~~
+We removed the requirement of using extended H2O jars. So in both manual and automatic mode of external backend you can
+download the corresponding H2O jars for extended backend via the script ``./bin/get-h2o-driver.sh``. This script
```

Comment:
`extended backend` -> `the external backend`?

Reply:
👍